src/auto/unix_fd_list.rs

```diff
@@ -0,0 +1,121 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use gio_sys;
+use glib;
+use glib::object::IsA;
+use glib::translate::*;
+use std::fmt;
+use std::mem;
+use std::ptr;
+
+glib_wrapper! {
```

Comment:
This whole struct sounds **very** unsafe. Shouldn't we handle instead types implementing `AsRawFD` instead of `i32`?

Reply:
Yes this needs to be manually implemented

Reply:
`RawFd` is just an alias to `c_int` — would be nice to use it, sure.

I'm not sure how `AsRawFd` offers *safety* advantages though. Convenience, yes — you don't have to convert to the raw fd manually.

The standard library considers `FromRawFd` unsafe, and `libc` functions that do stuff on fds are unsafe, but just passing them around shouldn't be a problem.

Reply:
Because that would prevent users to give whatever `i32` they want. We're providing a **safe** API, therefore we can't allow that.

Reply:
Slightly deter, not prevent ;)

```rust
use std::os::unix::io::{AsRawFd, RawFd};
struct Lol;
impl AsRawFd for Lol {
    fn as_raw_fd(&self) -> RawFd { 420_69 }
}
```

but sure, I'll implement that, for convenience

Reply:
Okay, now using `IntoRawFd` for `g_unix_fd_list_new_from_array` (since "file descriptors become the property of the new list and may no longer be used by the caller") and `RawFd` for the rest.